lib/config/configuration_test.go

```diff
@@ -419,6 +419,35 @@ func TestConfigReading(t *testing.T) {
 					},
 				},
 			},
+			AzureMatchers: []AzureMatcher{
+				{
+					Subscriptions:  []string{"sub1", "sub2"},
+					ResourceGroups: []string{"rg1", "rg2"},
```

Comment:
In real life, `rg1` would belong to only one of `sub1` or `sub2` (or maybe none for them). I will see how the implementation works out on this.

Reply:
It is as you noted the same scenario as when the resource group is not in the subscription at all. There is one api client per `Subscription x DB Type` and a fetcher is made for each tuple `Matcher x Subscription x DB Type x Resource Group`, using the appropriate api client for the subscription and DB type. When the fetcher tries to query for a resource group that does not exist in its subscription, it gets a "Not Found" error and that fetcher returns no results.

Reply:
That makes sense. Thanks for explaining.